pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -458,29 +459,32 @@ func TestUpdateNode(t *testing.T) {
 	}
 
 	for _, item := range testCases {
-		stopCh := make(chan struct{})
-		fakeClientset := fake.NewSimpleClientset(&v1.PodList{Items: item.pods})
-		controller, _, nodeIndexer := setupNewNoExecuteTaintManager(context.TODO(), fakeClientset)
-		nodeIndexer.Add(item.newNode)
-		controller.recorder = testutil.NewFakeRecorder()
-		go controller.Run(context.TODO())
-		controller.NodeUpdated(item.oldNode, item.newNode)
-		// wait a bit
-		time.Sleep(timeForControllerToProgress)
-		if item.additionalSleep > 0 {
-			time.Sleep(item.additionalSleep)
-		}
+		t.Run(item.description, func(t *testing.T) {
```

Comment:
Case 2: use the most commonly used pattern `t.Run()` to organize the tests, so we can defer to call `cancel()` with each testcase.